lib/plugins/aws/deploy/compile/events/apiGateway/lib/validate.js

```diff
@@ -4,80 +4,160 @@ const _ = require('lodash');
+  },
+
+  getHttpPath(http, functionName) {
+    if (typeof http.path === 'string') {
+      return http.path.replace(/^\//, '').replace(/\/$/, '');
+    }
+    const errorMessage = [
+      `Missing or invalid "path" property in function "${functionName}"`,
+      ' for http event in serverless.yml.',
+      ' If you define an http event, make sure you pass a valid value for it,',
+      ' either as string syntax, or object syntax.',
+      ' Please check the docs for more options.',
+    ].join('');
+    throw new this.serverless.classes.Error(errorMessage);
+  },
+
+  getHttpMethod(http, functionName) {
+    if (typeof http.method === 'string') {
+      const method = http.method.toLowerCase();
+
+      const allowedMethods = [
+        'get', 'post', 'put', 'patch', 'options', 'head', 'delete', 'any',
+      ];
+      if (allowedMethods.indexOf(method) === -1) {
+        const errorMessage = [
+          `Invalid APIG method "${http.method}" in function "${functionName}".`,
+          ` AWS supported methods are: ${allowedMethods.join(', ')}.`,
+        ].join('');
+        throw new this.serverless.classes.Error(errorMessage);
+      }
+      return method;
+    }
+    const errorMessage = [
+      `Missing or invalid "method" property in function "${functionName}"`,
+      ' for http event in serverless.yml.',
+      ' If you define an http event, make sure you pass a valid value for it,',
+      ' either as string syntax, or object syntax.',
+      ' Please check the docs for more options.',
+    ].join('');
+    throw new this.serverless.classes.Error(errorMessage);
+  },
+
+  getAuthorizer(http, functionName) {
+    const authorizer = http.authorizer;
+
+    let name;
+    let arn;
+    let identitySource;
+    let resultTtlInSeconds;
+    let identityValidationExpression;
+
+    if (typeof authorizer === 'string') {
+      if (authorizer.indexOf(':') === -1) {
+        this.serverless.service.getFunction(authorizer);
+
+        name = authorizer;
+
+        const normalizedName = name[0].toUpperCase() + name.substr(1);
+        arn = { 'Fn::GetAtt': [`${normalizedName}LambdaFunction`, 'Arn'] };
+      } else {
+        arn = authorizer;
+        const splitArn = arn.split(':');
+        const splitLambdaName = splitArn[splitArn.length - 1].split('-');
+        name = splitLambdaName[splitLambdaName.length - 1];
+      }
+    } else if (typeof authorizer === 'object') {
+      if (authorizer.arn) {
+        arn = authorizer.arn;
+        const splitArn = arn.split(':');
+        const splitLambdaName = splitArn[splitArn.length - 1].split('-');
+        name = splitLambdaName[splitLambdaName.length - 1];
+      } else if (authorizer.name) {
+        this.serverless.service.getFunction(authorizer.name);
+        const normalizedName = authorizer.name[0].toUpperCase() + authorizer.name.substr(1);
+        arn = { 'Fn::GetAtt': [`${normalizedName}LambdaFunction`, 'Arn'] };
+        name = authorizer.name;
```

Comment:
There seem to be some duplication here, could you maybe refactor it into one or two methods (ie `extractLambdaArn()` or something like that) and reuse that inside those if statements?